### ISSUE TITLE
[enhancement] Hybrid gunship + carrier modules

### ISSUE THREAD

**Issue by dtd98** [PavelZinchenko/event-horizon-frontier-builds#358]
Modules that empower both weapons and drones simultaneously

A drone carrier ship layout needs many green slots for factories and drone boosters, while a firepower-based layout often needs automatic re-loaders and high energy focuses. Trying to do both on the same ship tend to not work well because the space is limited.

Hybrid modules, that does both as a bundle, would enable hybrid firepower + carrier builds that would otherwise be sub-optimal. These modules would not outperform existing modules (re-loader, high energy focus, drone factory / boosters) on a pure gunship without drones, or a pure drone carrier, but they would be particularly efficient for a ship layout that involve being a hybrid of firepower and drone carrier aspects.

The modules could be something like the following:

[Integrated Offensive Unit]
(2x2 green, fusion of high-energy focus and drone damage booster effects)

Energy Consumption	+6
Damage			+50%
Energy Cost 			+50%
Drone Damage		+50%
Drone Build Time	+10%

[Integrated Logistics Unit]
(2x2 green, fusion of automatic re-loader and drone factory effects)

Energy Consumption	+10
Fire Rate 			+50%
Drone Build Time	5.0 seconds

[Integrated Radar System]
(2x2 green, fusion of rangemaster unit and drone navigation computer effects)

Energy Consumption	+6
Firing Range			+15%
Energy Cost			+15%
Drone range			+40%
Drone build time		+10% 

These modules can enable more build variety and allow drones to be effective on ships layouts that don't focus entirely on drones, and also possibly allow enemy ships layouts to canonically use them too (seems all enemy ships in EH have an actual module layout rather than hard-coded stats)

**Comment by Asmageddon** [comment]
I thought about it too, but it simply wouldn't be a good idea. It'd need to either be weak enough that nobody would use it, or it'd end up being too strong - especially lategame, when scientist upgrades mean that going from e.g. 50->55% of a single boost is a much smaller upgrade than going 40->44% on two.

**Comment by dtd98** [comment]
I agree its not ideal if its too strong that it would displace existing options on a ship that doesn't want both firepower boost and drone boosts.

Scientist stacking buffs on multiple aspects of the same module would make it overpowered, but it would not apply to the hybrid modules here, because 

for all of the existing modules:
- high energy focus
- automatic re-loader
- range master unit
- drone factory
- drone damage booster
- drone range booster

Those modules' primary effects are not upgradable by the scientist. So, for the hybrid modules, if their primary aspects remain non-upgradable they would not be overpowered in the late game.

![Untitled](https://github.com/user-attachments/assets/3061b6c2-fce9-47b9-aa25-c7da995182d1)
![Untitled2](https://github.com/user-attachments/assets/1f3a2a5a-ff25-4013-96f1-2aaa4cc8edeb)

See the scientist can only upgrade the trivial energy costs and mass for those mods.

Ideally a focused gunship layout would have no reason to use the hybrid modules over existing weapon ones, while a focused drone carrier layout also would have no reason to use the hybrid modules over existing drone ones. So if it were to only be significantly beneficial for hybrid gunship + carrier layouts which there is currently no optimal ways to set up, there would not be an issue where it can only be too weak or too strong. It can be not applicable to pure gunships and carriers, while at the same time highly efficient for hybrid builds. I believe that event horizon is also not that heavily balanced to such a fine extent like a competitive PvP based online game, there are more existing power level differences in existing stuff that makes many things sub-par in the late game, and hybrid modules like this would be far from a primary contributor to any imbalance.
